flake.nix

```diff
@@ -1,57 +1,101 @@
 {
-  description = "basic rust flake";
+  description = "adf-bdd, Abstract Dialectical Frameworks solved by Binary Decision Diagrams; developed in Dresden";
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
-    nixpkgs-unstable.url = "github:NixOS/nixpkgs/nixos-unstable";
     rust-overlay = {
       url = "github:oxalica/rust-overlay";
       inputs = {
         nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "flake-utils";
+        flake-utils.follows = "flake-utils/flake-utils";
       };
     };
-    flake-utils.url = "github:numtide/flake-utils";
-    gitignoresrc = {
-      url = "github:hercules-ci/gitignore.nix";
-      flake = false;
-    };
+    flake-utils.url = "github:gytis-ivaskevicius/flake-utils-plus";
   };
 
-  outputs = {
+  outputs = inputs @ {
     self,
-    nixpkgs,
-    nixpkgs-unstable,
     flake-utils,
-    gitignoresrc,
     rust-overlay,
     ...
-  } @ inputs:
-    {
-      #overlay = import ./nix { inherit gitignoresrc; };
-    }
-    // (flake-utils.lib.eachDefaultSystem (system: let
-      unstable = import nixpkgs-unstable {inherit system;};
-      pkgs = import nixpkgs {
-        inherit system;
-        overlays = [(import rust-overlay)];
-      };
-    in rec {
-      devShell = pkgs.mkShell {
-        RUST_LOG = "debug";
-        RUST_BACKTRACE = 1;
-        buildInputs = [
-          pkgs.rust-bin.stable.latest.rustfmt
-          pkgs.rust-bin.stable.latest.default
-          pkgs.rust-analyzer
-          pkgs.cargo-audit
-          pkgs.cargo-license
-          pkgs.cargo-tarpaulin
-          pkgs.cargo-kcov
-          pkgs.valgrind
-          pkgs.gnuplot
-          pkgs.kcov
-        ];
+  }:
+    flake-utils.lib.mkFlake {
+      inherit self inputs;
+      channels.nixpkgs.overlaysBuilder = channels: [rust-overlay.overlays.default];
+      outputsBuilder = channels: let
+        pkgs = channels.nixpkgs;
+        toolchain = pkgs.rust-bin.stable.latest.default;
+        platform = pkgs.makeRustPlatform {
+          cargo = toolchain;
+          rustc = toolchain;
+        };
+      in rec {
+        apps = rec {
+          adf-bdd = flake-utils.lib.mkApp {
+            drv = packages.adf-bdd;
+            exePath = "/bin/adf-bdd";
+          };
+        };
```

Comment:
You can just drop this, `nix run …#adb-bdd` will work as long as there is still an `adf-bdd` package with a `bin/adf-bdd` (otherwise you would need to set [`meta.mainProgram`](https://ryantm.github.io/nixpkgs/stdenv/meta/#var-meta-mainProgram)).